### PR TITLE
test: debug apex tests e2e tests

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
Implementation for E2E Debug Apex Tests

### What issues does this PR fix or reference?
@W-12370886@

See also https://salesforce.quip.com/qVEkABuGd7ao and https://salesforce.quip.com/q3tDAvAb2Udq

### CODE REVIEW

File: test/specs/debugApexTests.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { step } from 'mocha-steps';
+import {
+  CodeLens,
```
Comment (jeffb-sfdc) [inline test/specs/debugApexTests.e2e.ts:9]: CodeLens is not used and can be removed

File: test/specs/debugApexTests.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    expect(successNotificationWasFound).toBe(true);
+
+    // Continue with the debug session
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
+
+  step('Tear down and clean up the testing environment', async () => {
```
Comment (jeffb-sfdc) [inline test/specs/debugApexTests.e2e.ts:158]: All tests passed.  Nice!

File: test/scratchOrg.ts [hunk 3]
```diff
@@ -0,0 +1,343 @@
+/*
```
Comment (jeffb-sfdc) [inline test/scratchOrg.ts:1]: scratchOrg was renamed to testSetup.  test/scratchOrg.ts should not be part of your PR.

File: test/specs/debugApexTests.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    // Push source to scratch org
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Scratch Org and Override Conflicts', 5);
+  });
+
+  step('Debug All Tests via Apex Class', async () => {
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+
+    // Open an existing apex test (e.g. BotTest.cls, search for @isTest)
```
Comment (jeffb-sfdc) [inline test/specs/debugApexTests.e2e.ts:43]: I'm not seeing anything to do with "search for @isTest".  Is this a stale comment and this should be removed?

File: test/specs/debugApexTests.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
+
+  step('Tear down and clean up the testing environment', async () => {
+    await testSetup.tearDown();
+  });
+});
```
Comment (jeffb-sfdc) [inline test/specs/debugApexTests.e2e.ts:160]: The code uses break points.  Is it worth having tests which debug using check points as well?

File: test/utilities/apexUtils.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+import {
+  runCommandFromCommandPrompt
+} from './commandPrompt';
+import {
+  pause
+} from './miscellaneous';
+
+export async function createApexClassWithTest(name: string): Promise<void> {
+  const workbench = await browser.getWorkbench();
+  let textEditor: TextEditor;
```
Comment (jeffb-sfdc) [inline test/utilities/apexUtils.ts:20]: Please remove line 20 and add `const` to line 35

File: test/specs/debugApexTests.e2e.ts (hunk 4 above)
Comment (CristiCanizales) [reply test/specs/debugApexTests.e2e.ts:43]: yes. Thanks, good catch!

File: test/specs/debugApexTests.e2e.ts (hunk 5 above)
Comment (CristiCanizales) [reply test/specs/debugApexTests.e2e.ts:160]: In the trailApexReplayDebugger suite there's already a test that uses check points, so I think it's ok like this

**Review by jeffb-sfdc — APPROVED** [review]

**Pull request merged**